harness/tests/determined/pytorch/experimental/test_torch_batch_process.py

```diff
@@ -0,0 +1,627 @@
+        assert my_processor_instance.on_finish.call_count == 1
+
+
+@unittest.mock.patch(
+    "determined.pytorch.experimental._torch_batch_process._initialize_default_inference_context"
+)
+@unittest.mock.patch(
+    "determined.pytorch.experimental._torch_batch_process._synchronize_and_checkpoint"
+)
+@pytest.mark.parametrize(
+    "data_length, batch_size, checkpoint_interval, rank, num_slots, "
+    "max_batches, expected_process_batch_call_count",
+    [  # max_batches (2) < total batches (3) -> iterate for 2 batches
+        [50, 10, 1, 0, 2, 2, 2],
+        # max_batches (10) > total batches (3) -> iterate for 3 batches
+        [50, 10, 1, 0, 2, 10, 3],
+    ],
+    ids=["Valid max_batches arg", "Invalid max_batches arg"],
+)
+def test_torch_batch_process_only_use_valid_max_batches_arg(
+    mock_synchronize_and_checkpoint: unittest.mock.MagicMock,
+    mock_initialize_default_inference_context: unittest.mock.MagicMock,
+    data_length: int,
+    batch_size: int,
+    checkpoint_interval: int,
+    rank: int,
+    num_slots: int,
+    max_batches: int,
+    expected_process_batch_call_count: int,
+) -> None:
+    with test_util.set_mock_cluster_info(DEFAULT_ADDRS, rank, num_slots):
+        mock_initialize_default_inference_context.return_value = core._dummy_init(
+            distributed=_get_dist_context(rank=rank)
+        )
+
+        index_dataset = IndexData(data_length)
+
+        MyProcessorCLS = unittest.mock.Mock()
+        my_processor_instance = unittest.mock.Mock()
+        MyProcessorCLS.return_value = my_processor_instance
+
+        if max_batches > expected_process_batch_call_count:
+            # Warning is raised when max_batches > total number of batches
+            with pytest.warns(Warning):
+                experimental.torch_batch_process(
+                    dataset=index_dataset,
+                    batch_processor_cls=MyProcessorCLS,
+                    batch_size=batch_size,
+                    checkpoint_interval=checkpoint_interval,
+                    max_batches=max_batches,
+                )
+        else:
+            experimental.torch_batch_process(
+                dataset=index_dataset,
+                batch_processor_cls=MyProcessorCLS,
+                batch_size=batch_size,
+                checkpoint_interval=checkpoint_interval,
+                max_batches=max_batches,
+            )
+
+        assert my_processor_instance.process_batch.call_count == expected_process_batch_call_count
+        # on_finish should only be called once
+        assert my_processor_instance.on_finish.call_count == 1
+
+
+@unittest.mock.patch(
+    "determined.pytorch.experimental._torch_batch_process._initialize_default_inference_context"
+)
+@unittest.mock.patch(
+    "determined.pytorch.experimental._torch_batch_process._synchronize_and_checkpoint"
+)
+@unittest.mock.patch("determined.pytorch.experimental._torch_batch_process._reduce_metrics")
+def test_torch_batch_process_preemption(
```

Comment:
Similar comment here asking for a name that reflects the behavior and its outcome rather than just the feature.

Reply:
I've got a similar request about `test_torch_batch_process_reduce_metrics`, `test_torch_batch_process_invalid_checkpoint_interval`, `test_torch_batch_processor_context_to_device`, `test_torch_batch_processor_context_prepare_model_for_inference`, `test_torch_batch_processor_context_upload_path`, and `test_get_default_device`.